ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala

```diff
@@ -1100,5 +1100,45 @@ object LedgerApiErrors extends LedgerApiErrorGroup {
       )(implicit loggingContext: ContextualizedErrorLogger)
           extends LoggingTransactionErrorImpl(cause = s"Inconsistent: $details")
     }
+
+    @Explanation("Errors that arise from an internal system misbehavior.")
+    object Internal extends ErrorGroup() {
+      @Explanation(
+        "The participant didn't detect an inconsistent key usage in the transaction." +
+          "Within the transaction, an exercise, fetch or lookupByKey failed because " +
+          "the mapping of key -> contract id was inconsistent with earlier actions."
+      )
+      @Resolution("Contact support.")
+      object InternallyInconsistentKeys
+          extends ErrorCode(
+            id = "INTERNALLY_INCONSISTENT_KEYS",
+            ErrorCategory.SystemInternalAssumptionViolated, // Should have been caught by the participant
+          ) {
+        case class Reject(override val cause: String, _keyO: Option[GlobalKey] = None)(implicit
+            loggingContext: ContextualizedErrorLogger
+        ) extends LoggingTransactionErrorImpl(cause = cause) {
+          override def resources: Seq[(ErrorResource, String)] =
+            super.resources ++ _keyO.map(key => ErrorResource.ContractKey -> key.toString).toList
+        }
+      }
+
+      @Explanation(
+        "The participant didn't detect an attempt by the transaction submission " +
+          "to use the same key for two active contracts."
+      )
+      @Resolution("Contact support.")
+      object InternallyDuplicateKeys
+          extends ErrorCode(
+            id = "INTERNALLY_DUPLICATE_KEYS",
+            ErrorCategory.SystemInternalAssumptionViolated, // Should have been caught by the participant
+          ) {
+        case class Reject(override val cause: String, _keyO: Option[GlobalKey] = None)(implicit
+            loggingContext: ContextualizedErrorLogger
+        ) extends LoggingTransactionErrorImpl(cause = cause) {
+          override def resources: Seq[(ErrorResource, String)] =
+            super.resources ++ _keyO.map(key => ErrorResource.ContractKey -> key.toString).toList
+        }
+      }
+    }
```

Comment:
CC @fabiotudone-da Moved these to `LedgerApiErrors` since they can be returned by Sandbox-on-X as well.